src/box/alter.cc

```diff
@@ -1108,8 +1108,20 @@ CheckSpaceFormat::prepare(struct alter_space *alter)
 	if (new_space->upgrade != NULL) {
 		/*
 		 * Tuples stored in the space will be checked against
-		 * the new format during space upgrade.
+		 * the new format during space upgrade. We should only
+		 * check if format of indexed fields wasn't updated.
```

Comment:
I don't see any point in forbidding *compatible* space format updates (i.e. extending a space format). It should work both with space upgrade and without. This doesn't contradict the documentation.

Reply:
Allowed compatible updates.

Reply:
I meant dropping the first patch altogether - incompatible changes of indexed fields won't work anyway because it isn't possible to create such tuple format.

Reply:
Dropped the first patch.